server/routes/api/comments/comments.ts

```diff
@@ -360,4 +360,119 @@ router.post(
   }
 );
 
+router.post(
+  "comments.add_reaction",
+  rateLimiter(RateLimiterStrategy.TwentyFivePerMinute),
+  auth(),
+  feature(TeamPreference.Commenting),
+  validate(T.CommentsReactionSchema),
+  transaction(),
+  async (ctx: APIContext<T.CommentsReactionReq>) => {
+    const { id, emoji } = ctx.input.body;
+    const { user } = ctx.state.auth;
+    const { transaction } = ctx.state;
+
+    const comment = await Comment.findByPk(id, {
+      transaction,
+      rejectOnEmpty: true,
+      lock: {
+        level: transaction.LOCK.UPDATE,
+        of: Comment,
+      },
+    });
+    authorize(user, "addReaction", comment);
+
+    await Reaction.findOrCreate({
+      where: {
+        emoji,
+        userId: user.id,
+        commentId: id,
+      },
+      transaction,
+    });
+    const added = await comment.updateReactions({
+      type: "add",
+      emoji,
+      userId: user.id,
+      transaction,
+    });
+
+    if (added) {
+      await Event.createFromContext(
+        ctx,
+        {
+          name: "comments.add_reaction",
+          modelId: comment.id,
+          documentId: comment.documentId,
+          data: {
+            emoji,
+          },
+        },
+        { transaction }
+      );
+    }
+
+    ctx.body = {
+      success: true,
+    };
+  }
+);
+
+router.post(
+  "comments.remove_reaction",
+  rateLimiter(RateLimiterStrategy.TwentyFivePerMinute),
+  auth(),
+  feature(TeamPreference.Commenting),
+  validate(T.CommentsReactionSchema),
+  transaction(),
+  async (ctx: APIContext<T.CommentsReactionReq>) => {
+    const { id, emoji } = ctx.input.body;
+    const { user } = ctx.state.auth;
+    const { transaction } = ctx.state;
+
+    const comment = await Comment.findByPk(id, {
+      transaction,
+      rejectOnEmpty: true,
+      lock: {
+        level: transaction.LOCK.UPDATE,
+        of: Comment,
+      },
+    });
+    authorize(user, "removeReaction", comment);
```

Comment:
This permission is currently just checking the user and comment are in the same workspace, it's not enough – lets also check they can comment on the document.